src/core/ipfs/files.js

```diff
@@ -3,38 +3,68 @@
 const Importer = require('ipfs-unixfs-engine').importer
 const Exporter = require('ipfs-unixfs-engine').exporter
 const UnixFS = require('ipfs-unixfs')
+const bs58 = require('bs58')
+const through = require('through2')
+const isStream = require('isstream')
+const promisify = require('promisify-es6')
 
 module.exports = function files (self) {
   return {
-    add: (arr, callback) => {
-      if (typeof arr === 'function') {
-        callback = arr
-        arr = undefined
+    createAddStream: promisify((callback) => {
+      // TODO: wip
+      if (data === undefined) {
+        return new Importer(self._dagS)
       }
-      if (callback === undefined) {
-        callback = function noop () {}
+    }),
```

Comment:
noticed this now. @noffle why is this a TODO? 

This means that we won't be able to add folders recursively from the CLI. Saw now that this was also not added on js-ipfs-api